dotCMS/src/enterprise/java/com/dotcms/enterprise/priv/HostAssetsJobImpl.java

```diff
@@ -905,8 +943,25 @@ private void processToCopyContentletDependencies(final List<Contentlet> contents
                         records.add(c);
                     }
                 }
-                final ContentletRelationshipRecords related = new ContentletRelationships(
-                        destinationContent).new ContentletRelationshipRecords(r, true);
+				ContentletRelationshipRecords related;
+				if (sourceContent.getContentTypeId().equals(destinationContent.getContentTypeId())) {
+					related = new ContentletRelationships(
+							destinationContent).new ContentletRelationshipRecords(r, true);
+				} else {
+					final ContentTypeMapping contentTypeMapping =
+							copiedContentTypes.get(sourceContent.getContentTypeId());
+					final Field relationshipField =
+							APILocator.getContentTypeFieldAPI().byContentTypeIdAndVar(contentTypeMapping.destinationContentType.id(), r.getChildRelationName());
```

Comment:
rename r

Reply:
I didn't want to rename any more code in order to keep this PR as clean as possible. Those renames and further refactoring will be taken care of in another PR.